src/codeu/chat/util/Mergesort.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * A static class that mergesorts an ArrayList of Bookmarks. 
+ * 
+ * @author (Sarah Abowitz) 
+ * @version (V1.0.6 | 6.25.17)
+ */
+ 
+package codeu.chat.util;
+ 
+import codeu.chat.common.Bookmark;
+import codeu.chat.util.Time;
+ 
+import java.util.ArrayList;
+import java.util.Collections;
+ 
+public class Mergesort {
+  private ArrayList<Time> times;
+  private ArrayList<Time> timeHelper;
+  private ArrayList<Bookmark> bookmarks;
+  private ArrayList<Bookmark> properSequence;
+  private int number;
+  
+  public Mergesort() {}
+  
+  public ArrayList<Bookmark> sort(ArrayList<Time> values, ArrayList<Bookmark> marks){
+    times = values;
+    bookmarks = marks;
+    number = values.size();
+    timeHelper = new ArrayList<Time>(times);
+    int j = 0;
+    properSequence = new ArrayList<Bookmark>(number);
+    
+    Collections.sort(times);
+    orderBookmarks();
+    Collections.reverse(properSequence);
+    return properSequence;
+  }
+  
+  private Time getValue(int j){
+    return times.get(j);
+  }
+  
+  private void orderBookmarks(){
+    for(int i = 0; i< times.size(); i++){
+      orderBookmarksHelper(times.get(i));
```

Comment:
This method runs in O(n^2), because the `indexOf` method used in `orderBookmarksHelper` has to scan through the entire list to find the element it's searching for. 

Not necessary to optimize in this PR, but in the future, I'd suggest not reinventing the wheel :smile: , and just using `Collections.sort` to sort your list of Bookmarks by `Time`. You can do that by implementing a custom `Comparator`, not unlike what is done currently for other collections in [`Model.java`](https://github.com/vgomezASC/CodeU-Summer-2017/pull/11/files#diff-4c08b2e8ddca5684cd55d6070c2db1a9R49).